vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -138,4 +138,83 @@
       - MASTER_VM_1 in running_vms.list_vms
       - MASTER_VM_2 in running_vms.list_vms
     become: yes
-    become_user: root
\ No newline at end of file
+    become_user: root
+
+  # Start Unhealthy node testing
+  - name: Get KCP name
+    shell: |
+      kubectl get kcp -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
+      | .metadata.name'
+    register: kcp_name
+
+  - name: Define KCP_NAME variable
+    set_fact:
+      KCP_NAME: "{{ kcp_name.stdout }}"
+
+  - name: Scale KCP down to two replicas
+    shell: |
+      kubectl scale kcp "{{ KCP_NAME }}" -n "{{ NAMESPACE }}" --replicas=2
+
+  - name: Wait until KCP is scaled down and one node is Ready
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
+    retries: 200
+    delay: 20
+    register: ready_hosts
+    until: ready_hosts.stdout == "1"
+
+  - name: Mark "{{ WORKER_BMH }}" as unhealthy
+    shell: |
+      kubectl annotate bmh "{{ WORKER_BMH }}" -n "{{ NAMESPACE }}" capi.metal3.io/unhealthy=
+ 
+  - name: Delete worker Machine object "{{ WORKER_NODE }}"
+    shell: |
+      kubectl delete machine "{{ WORKER_NODE }}" -n "{{ NAMESPACE }}"
+
+  - name: Wait until one BMH is in ready state
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
+    retries: 200
+    delay: 20
+    register: ready_hosts
+    until: ready_hosts.stdout == "1"
+
+  - name: Wait until three BMH are provisioned
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioned | wc -l
+    retries: 200
+    delay: 20
+    register: provisioned_hosts
+    until: provisioned_hosts.stdout == "3"
+
+  - name: Get machinedeployment object name
+    shell: |
+      kubectl get machinedeployment -n metal3 -o json | jq -r '.items[]
+      | .metadata.name'
+    register: machinedeployment_name
+
+  - name: Define machine deployment
+    set_fact:
+      MACHINE_DEPLOYMENT_NAME: "{{ machinedeployment_name.stdout }}"
+
+  - name: Scale up the machinedeployment
+    shell: kubectl scale machinedeployment "{{ MACHINE_DEPLOYMENT_NAME }}" -n "{{ NAMESPACE }}" --replicas=2
+  
+  - pause:
+     minutes: 1
+
+  - name: Wait and verify that none of the nodes will not start provisioning
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioning | wc -l
+    retries: 10
+    delay: 20
+    register: provisioning_hosts
+    until: provisioning_hosts.stdout == "0"
+
+  - name: Remove unhealthy annotation from "{{ WORKER_BMH }}"
+    shell: |
+      kubectl annotate bmh "{{ WORKER_BMH }}" -n "{{ NAMESPACE }}" capi.metal3.io/unhealthy-
+    when: provisioning_hosts.stdout == "0"
```

Comment:
Maybe we should remove the condition here and instead rely on the task before this?

Reply:
Done